File: calculate_sum_even_or_odd.R
# The Time complexity of the function calculate_sum:
# 17+(2+10+3+3+1)n + 1 = 18 + 19n
calculate_sum <- function(x, n) {
  if (n<=0) return(1) #2

  odd <- n%%2 #2
  even <- 1 - odd #2

  s <- odd*1/x #3
  f <- odd + even*2 #3
  p <- odd*x + even*x*x #5

  for (i in 2:n) { #2
    f <- f * (2*i-odd*2)*(2*i-1) #10

    p <- p*x*x #3

    s <- s + f / p #3
  }

  return(s) #1
}

x <- 5
n <- 10

result <- calculate_sum(x, n)
cat("The result of the sum is:", result, "\n")

File: calculate_sum_even_or_odd.py
#D05
def calculate_sum(x, n):
    if n <= 0:
        return 1

    odd = n % 2
    even = 1 - odd

    s = odd * 1 / x
    f = odd + even * 2
    p = odd * x + even * x * x

    for i in range(2, n + 1):
        f = f * (2 * i - odd * 2) * (2 * i - 1)
        p = p * x * x
        s = s + f / p

    return s

x = 5
n = 11

result = calculate_sum(x, n)
print(f"The result of the sum is: {result}")

